plugins/module_utils/job.py

```diff
@@ -393,3 +393,109 @@ def _ddname_pattern(contents, resolve_dependencies):
             )
         )
     return str(contents)
+
+
+def FixDSName( name_in ):
+    result = []
+    ret_str = ""
+    dot_count = 0
+    seg_length = 0
+    prev_dot = False
+    add_paren = False
+    in_paren = False
+
+    # May not exceed 44 characters
+    if len(name_in) > 44:
+        return( None, "length" )
+
+    for c in name_in:
+        # slash become a paren for member section
+        if c == "/":
+            if not in_paren:
+                c = "("
+                add_paren = True
+            else:
+                return( None, "nested parentheses" )
+
+        # discard incoming escapes
+        if c == "\\":
+            continue
+
+        # check for special characters ('national characters') and hyphens
+        # manual testing shows escapes are not needed for # or -
+        if c in "@\$":
+            result.append("\\")
+
+        # There is a rule saying you can't have a - in member name, however,
+        # since we are supporting GDG, where you can pass a negative number,
+        # we are ignoring that rule.
+
+        if prev_dot:
+            # segment names must start with letter or national character
+            if c in "0123456789-":
+                return( None, "Bad segment start" )
+
+        if c == "(":
+            in_paren = True
+            result.append("\\")
+
+        if c == ")":
+            in_paren = False
+            result.append("\\")
+
+        if c == ".":
+            # '..' is not allowed in a dataset name
+            if prev_dot:
+                return( None, "Double Dot" )
+            else:
+                dot_count += 1
+                seg_length = 0
+                prev_dot = True
+
+                # slash only usable in last segment
+                if add_paren:
+                    return( None, "Slash only in last segment" )
+        else:
+            prev_dot = False
+            seg_length += 1
+            if seg_length > 8:
+                return( None, "Seg too long" )
+
+        result.append(c)
+
+    if add_paren:
+        result.append("\\")
+        result.append(")")
+
+    # There must be at least 2 name segments, so at least one '.'
+    if dot_count < 1:
+        return( None, "Too few segments" )
+
+    # DS Name may not end with a period
+    if c == '.':
+        return( None, "Ends with a dot" )
+
+    return (ret_str.join(result), None)
+
+
+def _dsname_escape(dsname_in):
```

Comment:
Also from the python style guide:  "_single_leading_underscore: weak “internal use” indicator. E.g. from M import * does not import objects whose names start with an underscore."

So it think FixDSName should have the single leading underscore and dsname_escape should not, since you import dsname_escape in zos_data_set module.

Reply:
yep... changed to _fix_ds_name, and dsname_escape...